src/aws-cpp-sdk-core/include/aws/core/client/AWSClient.h

```diff
@@ -183,6 +183,7 @@ namespace Aws
              */
             virtual void SetServiceClientName(const Aws::String& name);
 
+            void AppendToUserAgent(const Aws::String& valueToAppend);
```

Comment:
should we make it a r-value?

Reply:
I don't think it would bring much benefit here.
Unless you have an idea how it maybe more helpful here.